### PR TITLE
T128: User Account Frontend

### PR DESCRIPTION
**Pull request by jessicahe004**
## Description 

Feature implementation of the account front-end. Working on a form to allow users to manage their account. 

- For frontend work, you _must_ include screenshots of your work in the description.

(not yet there)

## Types of changes

What types of changes does your code introduce to WeWrap?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have made sure that eslint was passed locally
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: frontend/src/components/account.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import ReactDOM from 'react-dom/client';
+import { useState} from "react"; 
+import axios from 'axios';
+import { fchmod } from 'fs';
+import './signup.css';
+import { Link } from 'react-router-dom';
+
+export const Account = () => {
+
+
```
Comment (justinpchen94) [inline frontend/src/components/account.tsx]: nit: no more than 1 empty line in a row

File: frontend/src/components/account.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    const [warning, setWarning] = useState<string>('');
+    const [changePassword, setChangePassword] = useState<string>('');
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+
```
Comment (justinpchen94) [inline frontend/src/components/account.tsx]: generally it's not often people put the line break at the very top of the fn!

File: frontend/src/components/account.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        e.preventDefault(); 
+        await axios.post('http://localhost:8000/account/profile', {
+            changePassword
+        });
+    }
+
+    const resetPassword = async (e: React.FormEvent<HTMLFormElement>) => {
+
```
Comment (justinpchen94) [inline frontend/src/components/account.tsx]: same here

File: frontend/src/components/account.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    const [warning, setWarning] = useState<string>('');
+    const [changePassword, setChangePassword] = useState<string>('');
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+
+        e.preventDefault(); 
+        await axios.post('http://localhost:8000/account/profile', {
+            changePassword
+        });
```
Comment (justinpchen94) [inline frontend/src/components/account.tsx]: are these routes already defined and working or is this more a placeholder for now?

File: frontend/src/components/Account/useRemoveAccount.tsx
Comment (chinkevin25) [inline frontend/src/components/Account/useRemoveAccount.tsx:1]: i assume this file is not in use?

File: frontend/src/components/accountForm.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import { Link } from 'react-router-dom';
+import { EditProfile } from './Account/editProfileForm';
+import { LinkedAccounts } from './Account/linkedAccountsForm';
+import { EditPassword } from './Account/editPassword';
+import { Routes, Route } from 'react-router-dom';
+import {fetchAccountData} from './Account/fetchAccountData';
+
+export const Account = (): JSX.Element => {
+
+// const[data, loadingStatus] = fetchAccountData();
```
Comment (chinkevin25) [inline frontend/src/components/accountForm.tsx]: in general, lets try to remove code we know we don't need. So lets remove this

File: frontend/src/components/Account/editPassword.module.css [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+.editButton {
+    margin-left: 21em;
+}
+.buttonContainer{
+    flex-direction: row;
+}
+
+.profileNotifTitles{
```
Comment (chinkevin25) [inline frontend/src/components/Account/editPassword.module.css:70]: not a huge deal, but we don't really gain that much from saving a few chars. Might as well be explicit and call this `profileNotificationTitles`. In this case, notif, is pretty clear but just trying to build some better habits.

File: frontend/src/components/Account/editProfile.module.css [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    height: 781.36px;
+    display: flex;
+    background-color: #FFFFFF;
+    border-radius: 6px; 
+    flex-direction: column; 
+    border: 1px solid #D7D7D7;
+    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
+    box-sizing: border-box; 
+    align-items: center;
+    /* gap: 0.75em; */
```
Comment (chinkevin25) [inline frontend/src/components/Account/editProfile.module.css]: rm

File: frontend/src/components/Account/editProfile.module.css [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    height: 46.25px;
+    text-align: center;
+    font-size: 10px;
+    display: flex; 
+    align-items: center;
+}
+
+.editButton, .cancelButton, .saveButton {
+    width: 142.79px; 
+    height: 38.95px;
```
Comment (chinkevin25) [inline frontend/src/components/Account/editProfile.module.css:31]: fractional pixels are interesting they are allowed, but just want to highlight that it's a little inconsistent. TLDR it depends on zoom level, and browsers. Obviously 1 px shift any direction isn't really that important in many cases. 
I assume you got this number from figma though so shame on ray! 

https://medium.com/kajabi-ux/css-and-sub-pixel-rendering-the-case-of-the-clipped-border-4652c5a1b5ab
https://stackoverflow.com/questions/34676263/sub-pixels-calculated-and-rendered-differently-among-browsers

No required changes, but just want you to understand the sub-pixel problem

File: frontend/src/components/Account/editProfile.module.css [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+.editButton, .saveButton {
+    border: 1px solid #EF476F;
+    background-color: #EF476F;
+    color: white;
+}
+
+.editButton {
+    margin-left: 14em;
```
Comment (chinkevin25) [inline frontend/src/components/Account/editProfile.module.css:44]: this large of an `em` is interesting to me, I assume you're doing this to offset it from the text a certain distance away, but maybe some percent value makes more sense? Maybe even a `position` css modifier? No change needed though

**Review by chinkevin25 — APPROVED** [review]

File: frontend/src/components/account.tsx (hunk 4 above)
Comment (jessicahe004) [reply frontend/src/components/account.tsx]: they are placeholders for now

File: frontend/src/components/Account/useRemoveAccount.tsx
Comment (jessicahe004) [reply frontend/src/components/Account/useRemoveAccount.tsx:1]: i'll remove that

**Pull request merged**